articles/tutorials/qft-tutorial.md

```diff
@@ -0,0 +1,624 @@
+	For now there is nothing returned, but later on we will see an example of assigning and processing these values.
+	With the resultant data in our hands and totally classical, we can do whatever we'd like with it.
+
+Your full `host.py` file should be this:
+
+```python
+import qsharp
+from Quantum.Operations import Perform3qubitQFT
+
+Perform3qubitQFT.simulate()
+```
+
+Run the Python file, and printed in your console you should see the `Message` and `DumpMachine` outputs below. 
+
+
+#### [C#](#tab/tabid-csharp)
+
+Following the same [instructions](xref:microsoft.quantum.howto.createproject) as above, create a C# host file, and rename it to `host.cs`.
+
+The C# host has four parts:
+1. Construct a quantum simulator.
+	In the code below, this is the variable `qsim`.
+2. Compute any arguments required for the quantum algorithm.
+	There are none in this example.
+3. Run the quantum algorithm. 
+	Each Q# operation generates a C# class with the same name. 
+	This class has a `Run` method that **asynchronously** executes the operation.
+	The execution is asynchronous because execution on actual hardware will be asynchronous. 
+	Because the `Run` method is asynchronous, we call the `Wait()` method; this blocks execution until the task completes and returns the result synchronously. 
+4. Process the returned result of the operation.
+	For now, the operation returns nothing.
+
+
+```csharp
+using System;
+
+using Microsoft.Quantum.Simulation.Core;
+using Microsoft.Quantum.Simulation.Simulators;
+
+namespace Quantum.Operations
+{
+    class Driver
+    {
+        static void Main(string[] args)
+        {
+            using (var qsim = new QuantumSimulator())
+            {
+                Perform3QubitQFT.Run(qsim).Wait();
+            }
+            
+            System.Console.WriteLine("Press any key to continue...");
+            Console.ReadKey();
+        }
+    }
+}
+
+```
+Run the application, and your output should match that below.
+The program will exit after you press a key.
+***
+
+```Output
+Initial state |000>:
+# wave function for qubits with ids (least to most significant): 0;1;2
+|0>:	 1.000000 +  0.000000 i	 == 	******************** [ 1.000000 ]     --- [  0.00000 rad ]
+|1>:	 0.000000 +  0.000000 i	 == 	                     [ 0.000000 ]                   
+|2>:	 0.000000 +  0.000000 i	 == 	                     [ 0.000000 ]                   
+|3>:	 0.000000 +  0.000000 i	 == 	                     [ 0.000000 ]                   
+|4>:	 0.000000 +  0.000000 i	 == 	                     [ 0.000000 ]                   
+|5>:	 0.000000 +  0.000000 i	 == 	                     [ 0.000000 ]                   
+|6>:	 0.000000 +  0.000000 i	 == 	                     [ 0.000000 ]                   
+|7>:	 0.000000 +  0.000000 i	 == 	                     [ 0.000000 ]                   
+After:
+# wave function for qubits with ids (least to most significant): 0;1;2
```

Comment:
"wavefunction" should be one word.

Reply:
This is in generated output, and accurately reflects what is printed by the `DumpMachine` and `DumpRegister` callables; the fix "wave function" → "wavefunction" should thus be applied to this line: https://github.com/microsoft/qsharp-runtime/blob/ebb4b2a9de0110ab45a6e38b38cfbd664759483f/src/Simulation/Simulators/QuantumSimulator/Dump.cs#L31.

Reply:
Made the change in runtime repo and merged a PR to master there.

Reply:
Btw, in my "copyedit" PR, I noticed the inconsistent use of `wave function` versus `wavefunction` throughout the current docs, but didn't change it. For the scientific concept, English uses [wave function](https://en.wikipedia.org/wiki/Wave_function) two words; I assumed "wavefunction" one word might be some Q#-specific term of art; I don't know enough about the domain to know when the one-word version is appropriate.